icu4c/source/i18n/smpdtfmt.cpp

```diff
@@ -1074,7 +1074,8 @@ SimpleDateFormat::_format(Calendar& cal, UnicodeString& appendTo,
     UDisplayContext capitalizationContext = getContext(UDISPCTX_TYPE_CAPITALIZATION, status);
 
     // loop through the pattern string character by character
-    for (int32_t i = 0; i < fPattern.length() && U_SUCCESS(status); ++i) {
+    int32_t len = fPattern.length();
```

Comment:
Maybe `patternLength` instead of just `len`?

Reply:
Thanks, fixed.